src/wallet/wallet2.h

```diff
@@ -2608,7 +2629,16 @@ namespace boost
       a & x.key_images;
       a & x.tx_key;
       a & x.dests;
-      a & x.construction_data;
+      if (ver < 4)
+      {
+        tools::wallet2::tx_construction_data pre_carrot_construction_data;
+        a & pre_carrot_construction_data;
+        x.construction_data = pre_carrot_construction_data;
```

Comment:
Is both serialization/de-serialization tested for this? Seems like this would only work for de-serialization

Also missing loading `subaddr_account` and `subaddr_indices` for `ver < 4` (which makes sense, just inconsistent with the other serialization definition)

Reply:
Since `pending_tx`'s Boost class version is set to 4, AFAIK, we won't ever be *serializing* when `ver < 4`, only *deserializing*, but please correct me if wrong.

Reply:
You're right about `subaddr_account` and `subaddr_indices`, though, thanks!

Reply:
Sounds ok to me, maybe just a comment explaining that would be good